pkg/defaults/provider.go

```diff
@@ -0,0 +1,198 @@
+package defaults
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/gosimple/slug"
+)
+
+const (
+	k0sBinsSubDirDarwin = "Library/Caches/k0sctl/k0s/linux/amd64"
+	k0sBinsSubDirLinux  = ".cache/k0sctl/k0s/linux/amd64"
+)
+
+// NewProvider returns a new DefaultsProvider using the provided base dir.
+// Base is the base directory inside which all the other directories are
+// created.
+func NewProvider(base string) *DefaultsProvider {
+	obj := &DefaultsProvider{Base: base}
+	obj.Init()
+	return obj
+}
+
+// DefaultsProvider is an entity that provides default values used during
+// HelmVM installation.
+type DefaultsProvider struct {
+	Base string
+}
+
+// Init makes sure all the necessary directory exists on the system.
+func (d *DefaultsProvider) Init() {
+	if err := os.MkdirAll(d.K0sctlBinsSubDir(), 0755); err != nil {
+		panic(fmt.Errorf("unable to create basedir: %w", err))
```

Comment:
I'm not a massive fan of using `panic()` like this when an end user is going to be seeing the messages. it's nicer to use something like `logrus.Fatalf()`

Reply:
Yeah, that makes sense. Let me refactor that to replace the calls by `logrus.Fatalf`.

Reply:
@danj-replicated solved, can you please take a look ?